File: dotnet_backend/src/AIMedical.Api/Models/Entities/Patient.cs
namespace AIMedical.Api.Models.Entities;

public class Patient
{
    public int Id { get; set; }
    public int TenantId { get; set; }
    public string PatientUid { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Mobile { get; set; }
    public string? Nid { get; set; }
    public DateTime? DateOfBirth { get; set; }
    public string? Gender { get; set; }
    public string? BloodGroup { get; set; }
    public string? Address { get; set; }
    public string? EmergencyContactName { get; set; }
    public string? EmergencyContactPhone { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation
    public Tenant? Tenant { get; set; }
    public ICollection<Visit> Visits { get; set; } = [];
    public ICollection<Appointment> Appointments { get; set; } = [];
}

File: backend/app/schemas/patient.py
from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel


class PatientBase(BaseModel):
    name: str
    mobile: Optional[str] = None
    nid: Optional[str] = None
    dob: Optional[date] = None
    gender: Optional[str] = None
    blood_group: Optional[str] = None
    address: Optional[str] = None
    emergency_contact_name: Optional[str] = None
    emergency_contact_phone: Optional[str] = None


class PatientCreate(PatientBase):
    tenant_id: int


class PatientUpdate(BaseModel):
    name: Optional[str] = None
    mobile: Optional[str] = None
    nid: Optional[str] = None
    dob: Optional[date] = None
    gender: Optional[str] = None
    blood_group: Optional[str] = None
    address: Optional[str] = None
    emergency_contact_name: Optional[str] = None
    emergency_contact_phone: Optional[str] = None
    is_active: Optional[bool] = None


class PatientResponse(PatientBase):
    id: int
    tenant_id: int
    patient_uid: str
    is_active: bool
    created_at: datetime

    class Config:
        from_attributes = True


class PatientListResponse(BaseModel):
    total: int
    items: list[PatientResponse]


class AllergyBase(BaseModel):
    allergen: str
    severity: Optional[str] = None
    reaction: Optional[str] = None


class AllergyCreate(AllergyBase):
    pass


class AllergyResponse(AllergyBase):
    id: int
    patient_id: int
    noted_by_id: Optional[int] = None
    noted_at: datetime

    class Config:
        from_attributes = True


class DuplicateCheckRequest(BaseModel):
    mobile: Optional[str] = None
    name: str
    dob: Optional[date] = None


class MergeRequestCreate(BaseModel):
    source_patient_id: int
    target_patient_id: int
    reason: Optional[str] = None
